### PR TITLE
Reverting the anti-affinity from CoreDNS pods

### PR DESCRIPTION
**Pull request by rajansandeep**
**What this PR does / why we need it**:
Following #54164 and #59357, removing the anti-affinity from CoreDNS.
 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rajansandeep** [comment]
cc @bowei @MrHohn

File: cluster/addons/dns/coredns.yaml.base [hunk 1]
```diff
@@ -98,21 +98,9 @@ spec:
...
-                matchExpressions:
-                - key: k8s-app
-                  operator: In
-                  values:
-                  - coredns
-                topologyKey: kubernetes.io/hostname
       containers:
       - name: coredns
-        image: coredns/coredns:1.0.4
+        image: coredns/coredns:1.0.6
```
Comment (MrHohn) [inline cluster/addons/dns/coredns.yaml.base]: It seems like this is doing more than just reverting the anti-affinity. Would be great if we could split them or add a better commit message.

**Comment by timothysc** [comment]
We are in bug-fix only mode, so it would need to specifically be just the fix.

**Comment by rajansandeep** [comment]
@timothysc @MrHohn ok, So if the PR only removes the pod anti-affinity it will be in 1.10?
Note: I included the version and manifest update to be in sync with kubeadm.

**Comment by MrHohn** [comment]
> ok, So if the PR only removes the pod anti-affinity it will be in 1.10?
Note: I included the version and manifest update to be in sync with kubeadm.

@rajansandeep Reverting the "pod anti-affinity" could certainly be in 1.10. I have no objections for the manifest sync as well if that is a required fix. Though took a quick look it shows kubeadm is using coreDNSVersion `1.0.4`?
https://github.com/kubernetes/kubernetes/blob/2853738a6b2710b7627b82492c76d10c8046377f/cmd/kubeadm/app/phases/addons/dns/versions.go#L26

**Comment by rajansandeep** [comment]
@MrHohn Ah yes, the changes to the manifest in kubeadm is in a separate PR which is on-course to be merged for 1.10.

https://github.com/kubernetes/kubernetes/pull/58828/files#diff-6f6577e723201d3cf7ff85de45a73068R26

**Review by timothysc — APPROVED** [review]
/approve with the changes outlined to kubeadm.  Still requires final lgtm and approval from the other area.

**Comment by rajansandeep** [comment]
@xiangpengzhao @MrHohn we need the milestone tag and labels for the PR to go through?

**Comment by MrHohn** [comment]
@rajansandeep Sorry I don't have the privilege to apply milestone. @timothysc might be able to help?

**Comment by rajansandeep** [comment]
@timothysc please provide the milestone and labels for the PR to get merged in 1.10. 
Thank you!

**Pull request merged**